sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -221,8 +221,9 @@ int gnrc_ipv6_nib_get_next_hop_l2addr(const ipv6_addr_t *dst,
         unsigned iface = (node == NULL) ? 0 : _nib_onl_get_if(node);
 
         if ((node != NULL) || _on_link(dst, &iface)) {
-            DEBUG("nib: %s is on-link or in NC, start address resolution\n",
-                  ipv6_addr_to_str(addr_str, dst, sizeof(addr_str)));
+            DEBUG("nib: %s is %s, start address resolution\n",
+                  ipv6_addr_to_str(addr_str, dst, sizeof(addr_str)),
+                  node ? "in NC" : "on-link");
```

Comment:
If an entry is in the NC it is by definition on-link. Not sure why the distinction is made in this debug message...

Reply:
I just found it useful for debugging to know whether `_on_link()` returned true or if `node` was not NULL.